### PR TITLE
Food refactor, Trash states for drinks and updated icons

### PR DESCRIPTION
**Pull request by alsoandanswer**
Title.

I had to slay some beasts such as the feeder eater shit in snacks.dm.

I ported over some stuff that I was able to refactor into food.dm.

Also added a new dropsound - paper cups. (and tweaked the vending coffee sound to be a lil' faster)

This probably paves the way for future drink trash states as well. But for now this is good enough for me.

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

**Comment by Geevies** [comment]
Feedback thread for the new sprites.

**Comment by alsoandanswer** [comment]
Aight

File: code/modules/reagents/reagent_containers/food.dm [hunk 1]
```diff
@@ -2,19 +2,35 @@
...
-	drop_sound = 'sound/items/drop/food.ogg'
-	var/is_liquid = FALSE // Handles whether you can eat with a fork & which eating sound is played
-
-/obj/item/reagent_containers/food/self_feed_message(var/mob/user)
-	to_chat(user, "<span class='notice'>You [is_liquid? "drink" : "eat"] \the [src].</span>")
+	var/trash = null
+	var/is_liquid = TRUE
 	
+/obj/item/reagent_containers/food/self_feed_message(var/mob/user)
+	to_chat(user, "<span class='notice'>You [is_liquid? "drink from" : "eat"] \the [src].</span>")
```
Comment (NonQueueingMatt) [inline code/modules/reagents/reagent_containers/food.dm]:
```suggestion
	to_chat(user, "<span class='notice'>You [is_liquid ? "drink from" : "eat"] \the [src].</span>")
```

File: code/modules/reagents/reagent_containers/food.dm [hunk 2]
```diff
@@ -2,19 +2,35 @@
...
+	if(!reagents.total_volume)
+		if(!user)
+			user = target
+		user.drop_from_inventory(src)	//so trash actually stays in the active hand.
+		if(trash)
+			if(ispath(trash,/obj/item))
+				var/obj/item/TrashItem = new trash(user)
+				user.put_in_hands(TrashItem)
+			else if(istype(trash,/obj/item))
+				user.put_in_hands(trash)
```
Comment (NonQueueingMatt) [inline code/modules/reagents/reagent_containers/food.dm]: What is this even for? Just make sure they're all types.

File: code/modules/reagents/reagent_containers/food.dm [hunk 3]
```diff
@@ -2,19 +2,35 @@
...
+/obj/item/reagent_containers/food/self_feed_message(var/mob/user)
+	to_chat(user, "<span class='notice'>You [is_liquid? "drink from" : "eat"] \the [src].</span>")
 
 /obj/item/reagent_containers/food/feed_sound(var/mob/user)
 	if(is_liquid)
 		playsound(user.loc, 'sound/items/drink.ogg', rand(10, 50), 1)
 	else
 		playsound(user.loc, 'sound/items/eatfood.ogg', rand(10, 50), 1)
+
+/obj/item/reagent_containers/food/proc/on_consume(var/mob/user = null, var/mob/target)
```
Comment (NonQueueingMatt) [inline code/modules/reagents/reagent_containers/food.dm]:
```suggestion
/obj/item/reagent_containers/food/proc/on_consume(var/mob/user, var/mob/target)
```

File: code/modules/reagents/reagent_containers/food.dm [hunk 4]
```diff
@@ -2,19 +2,35 @@
...
+	if(!reagents.total_volume)
+		if(!user)
+			user = target
```
Comment (NonQueueingMatt) [inline code/modules/reagents/reagent_containers/food.dm]: What's this for?

Comment (alsoandanswer) [reply code/modules/reagents/reagent_containers/food.dm]: I'm pretty sure it's a sanity check on whether the person feeding (user) and the eater (target) both exist, as a failsafe if the feeder isn't there.

I'll comment it out and see how it goes

File: code/modules/reagents/reagent_containers/food.dm (hunk 2 above)
Comment (alsoandanswer) [reply code/modules/reagents/reagent_containers/food.dm]: I have no idea. This is oldcode.

File: code/modules/reagents/reagent_containers/food.dm [hunk 5]
```diff
@@ -2,19 +2,35 @@
...
+		if(trash)
+			if(ispath(trash,/obj/item))
+				var/obj/item/TrashItem = new trash(user)
+				user.put_in_hands(TrashItem)
+			else if(istype(trash,/obj/item))
+				user.put_in_hands(trash)
+			target.visible_message(span("notice", "[target] finishes [is_liquid ? "drinking" : "eating"] \the [src]."),
+								   span("notice","You finish [is_liquid ? "drinking" : "eating"] \the [src]."))
+		qdel(src)
+	return
```
Comment (skull132) [inline code/modules/reagents/reagent_containers/food.dm]:
```suggestion
```

**Comment by alsoandanswer** [comment]
!review

**Pull request merged**